### ISSUE TITLE
feature suggestion: stretch/shrink chapter timecodes

### ISSUE THREAD

**Issue by zootie** [mbunkus/mkvtoolnix#1609]
In the GUI shell, in the "chapter mass modification" dialog, add an option to multiply timecodes by a given factor, to stretch/shrink all chapter timecodes. This would allow to transplant an existing set of chapters to a different edition of the same video (which only has a difference in time code positions, progressively changing in time).

There is already a constrict/expand option, but it only changes parent/child chapter bounds to the provided timecodes to be consistent (so they are contained and don't overlap with others). There is also a static shift option (to move chapters by a given unchanging offset).

And/or, but probably more complicated (yet consistent with other elements), allow to change chapter timecodes in the merge UI (as with audio and subtitle tracks, enable/implement relevant parameters from the "Timecodes and default duration" group)

Thank you for a great project!

**Comment by zootie** [comment]
It seems that a chapter stretch feature would be close to the existing shift functionality (in the mass chapter modification dialog) and likely simple to implement. Rather than adding/substracting, just have to multiply by the given value. Alas, I'm not familiar enough with the code and tools used to attempt to make the changes myself.

**Comment by mbunkus** [comment]
Thanks for the suggestion. I'll implement a multiplicative operation in the GUI for that. I won't implement such operations for the command line version (and therefore not in the merge tool in the GUI either), though.

**Comment by zootie** [comment]
I figured it might be a bit too complicated for the merge tool command line. It would be more consistent with the rest of the UI for it, but chapters seem to be a different animal. As long as we can do it in the chapter editor GUI, it would allow for synchronizing different editions.

In case it is useful as test data. I did this one manually (repositioned the chapter timecodes by exporting it as simple, multiplying in Excel by 1.042609248, and then pasting back one by one into the chapter XML editor (Excel formatting seems to only allow 2 digits for milliseconds, so there is a bit of an error). In this one, I'm converting chapters from DVD edition of Les Misérables (2012) to the BluRay edition (which only differs in timecodes).

[Les Miserables.21.renamed.chapters.xml.txt](https://github.com/mbunkus/mkvtoolnix/files/144150/Les.Miserables.21.renamed.chapters.xml.txt)

[Les Miserables.21.repositioned.renamed.chapters.xml.txt](https://github.com/mbunkus/mkvtoolnix/files/144151/Les.Miserables.21.repositioned.renamed.chapters.xml.txt)
